### PR TITLE
dev: Use watchman for watching go files changes

### PR DESCRIPTION
**Pull request by keegancsmith**
Chokidar would open too many files on my laptop. Additionally I already use
watchman to speed up my git operations. Watchman is also fast enough to watch
all go files, instead of relying on our "godirs" hacks.

### CODE REVIEW

File: dev/changewatch.sh [hunk 1]
```diff
@@ -24,23 +24,38 @@ useChokidar() {
...
+    echo >&2 "Using watchman."
+    set -e
+    pushd dev/watchmanwrapper
+    go build
+    popd
+    exec dev/watchmanwrapper/watchmanwrapper dev/handle-change.sh <<-EOT
+["subscribe", ".", "gochangewatch", {
+  "expression": ["anyof",
+    ["suffix", "go"],
+    ["dirname", "cmd/symbols"],
```
Comment (mrnugget) [inline dev/changewatch.sh]: Why `cmd/symbols` specifically?

Comment (keegancsmith) [reply dev/changewatch.sh]: look at how we call chokidar above. Anything that changes in cmd/symbols requires rebuilding a docker image.

Comment (mrnugget) [reply dev/changewatch.sh]: Ah! Those two lines were the only ones GitHub folded away 🤦‍♂️

**Comment by keegancsmith** [comment]
@sqs @beyang (and any other linux users). Do you use inotifywait? Would you consider switching to watchman so I can remove support for inotify. As it stands, I can see the patterns are out of date anyways (for symbols service)

**Comment by tsenart** [comment]
Would this integrate with `./enterprise/dev/start.sh`?

**Comment by keegancsmith** [comment]
> Would this integrate with `./enterprise/dev/start.sh`?

yup. that just shells out to this script. In fact we could get rid of that script since all that script does is give extra dirs to watch.

**Comment by tsenart** [comment]
I'm having trouble compiling watchman on Ubuntu: https://github.com/facebook/watchman/issues/638

**Comment by tsenart** [comment]
Solved it.

File: dev/watchmanwrapper/main.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package main
```
Comment (tsenart) [inline dev/watchmanwrapper/main.go]: Can you add a comment saying what this wrapper is for?

File: dev/changewatch.sh [hunk 3]
```diff
@@ -24,23 +24,38 @@ useChokidar() {
...
 }
 
-useInotifywrapper() {
-	echo >&2 "Using inotifywrapper."
-	exec dev/inotifywrapper/inotifywrapper $(dirs_path $GO_DIRS) \
-		-match '\.go$' \
-		-match 'cmd/frontend/graphqlbackend/schema\.graphql' \
-		-match 'schema/.*.json' \
-		-cmd './dev/handle-change.sh'
+execInotifywrapper() {
```
Comment (felixfbecker) [inline dev/changewatch.sh]: Is this still used / is there any value in still keeping this around?

**Comment by felixfbecker** [comment]
Consider adding a note to our dev setup guide to `brew install watchman`

**Pull request merged**